### PR TITLE
Enhance <TextInputRow> for custom rendering and textarea autosizing

### PR DESCRIPTION
**Pull request by tz5514**
# Purpose
Enhance `<TextInputRow>` for custom rendering and textarea autosizing.

# Changes
### Changed
- [Form] Enhance `<TextInputRow>`:
    - Can render custom input by `renderInput` or `inputComponent` prop.
    - Can specify textarea autosize rules by `minRows` and `maxRows` props.

# Risk
none.

# TODOs
none.

### CODE REVIEW

File: packages/form/src/TextInputRow.js [hunk 1]
```diff
@@ -95,55 +70,51 @@ class TextInputRow extends PureComponent {
...
         }
 
-        return (
-            <input
-                type="text"
-                onChange={this.props.onChange}
-                {...sharedProps}
-                {...inputProps} />
+        return React.createElement(
+            isFunction(inputComponent) ? inputComponent : 'input',
```
Comment (zhusee2) [inline packages/form/src/TextInputRow.js]: Maybe you can just set default value of `'input'` to `props.inputComponent`? And allow it to be either a String or a Function.

File: packages/form/src/TextInputRow.js [hunk 2]
```diff
@@ -95,55 +70,51 @@ class TextInputRow extends PureComponent {
...
 
-        if (multiLine) {
-            return (
-                <textarea
-                    style={textareaStyle}
-                    onChange={this.handleInputChange}
-                    {...sharedProps}
-                    {...inputProps} />
-            );
+        if (isFunction(renderInput)) {
```
Comment (zhusee2) [inline packages/form/src/TextInputRow.js]: What if it's used this way:

```jsx
<TextInputRow
    multiLine
    renderInput={func}
/>
```

Do you intent to let `renderInput` overridden by `multiLine`?

File: packages/form/src/__tests__/TextInputRow.test.js [hunk 3]
```diff
@@ -86,45 +88,20 @@ describe('Pure <TextInputRow>', () => {
...
-        // componentDidMount()
-        expect(controlledWrapper.find('textarea').prop('style')).toEqual({ height: 30 });
-
-        // componentDidUpdate()
-        controlledWrapper.setProps({ value: 'more bar' });
-        controlledWrapper.update();
-        expect(controlledWrapper.find('textarea').prop('style')).toEqual({ height: 60 });
+        const textareaWrapper = uncontrolledWrapper.find(AutoSizeTextarea);
+        expect(textareaWrapper.exists()).toBeTruthy();
+        expect(textareaWrapper.props()).toMatchObject({ minRows: 5, maxRows: 10 });
```
Comment (zhusee2) [inline packages/form/src/__tests__/TextInputRow.test.js]: 👍 nice shallow testing

File: packages/form/src/TextInputRow.js (hunk 1 above)
Comment (tz5514) [reply packages/form/src/TextInputRow.js]: Sounds great!

File: packages/form/src/TextInputRow.js (hunk 2 above)
Comment (tz5514) [reply packages/form/src/TextInputRow.js]: No, input can't be customize when `multiLine` mode.

**Pull request merged**